File: 5122.java
import java.io.*;
import java.util.*;

class Solution {
	public static void main(String args[]) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = null;
		StringBuilder sb = new StringBuilder();

		int testCase = Integer.parseInt(br.readLine());

		for (int tc = 1; tc <= testCase; tc++) {
			int listNumber, commandNumber, printIndex; // 수열의 길이, 추가 횟수, 출력할 인덱스 번호
			LinkedList<Integer> numberList = new LinkedList<>(); // 수열

			st = new StringTokenizer(br.readLine(), " "); // 수열의 길이, 추가 횟수, 출력할 인덱스 번호 입력
			listNumber = Integer.parseInt(st.nextToken());
			commandNumber = Integer.parseInt(st.nextToken());
			printIndex = Integer.parseInt(st.nextToken());

			st = new StringTokenizer(br.readLine(), " "); // 수열 입력
			while (st.hasMoreTokens()) {
				numberList.add(Integer.parseInt(st.nextToken()));
			}

			for (int i = 0; i < commandNumber; i++) {
				st = new StringTokenizer(br.readLine(), " ");

				String command = st.nextToken();

				if (command.equals("I")) { // 추가
					int index = Integer.parseInt(st.nextToken());
					int numberToAdd = Integer.parseInt(st.nextToken());

					numberList.add(index, numberToAdd);
				}

				else if (command.equals("D")) {
					int index = Integer.parseInt(st.nextToken());

					numberList.remove(index);
				}

				else {
					int index = Integer.parseInt(st.nextToken());
					int numberToChange = Integer.parseInt(st.nextToken());

					numberList.set(index, numberToChange);
				}
			}

			int answer = -1;

			if (numberList.size() >= printIndex) {
				answer = numberList.get(printIndex);
			}

			sb.append("#").append(tc).append(" ").append(answer).append("\n");
		}

		System.out.println(sb);
		br.close();
	}
}

File: 5122.py
test_case = int(input())
result = []

for tc in range(1, test_case + 1):
    list_number, command_number, print_index = map(int, input().split())

    number_list = list(map(int, input().split()))

    for _ in range(command_number):
        command = input().split()

        if command[0] == "I":  # Insert
            idx, number_to_add = int(command[1]), int(command[2])
            number_list.insert(idx, number_to_add)

        elif command[0] == "D":  # Delete
            idx = int(command[1])
            if 0 <= idx < len(number_list):
                number_list.pop(idx)

        else:  # Change
            idx, number_to_change = int(command[1]), int(command[2])

            if 0 <= idx < len(number_list):
                number_list[idx] = number_to_change

    answer = number_list[print_index] if print_index < len(number_list) else -1
    result.append(f"#{tc} {answer}")

print("\n".join(result))
